src/main/java/org/opentripplanner/routing/graphfinder/PlaceFinderTraverseVisitor.java

```diff
@@ -80,6 +81,11 @@ public PlaceFinderTraverseVisitor(
     includeVehicleRentals = shouldInclude(filterByPlaceTypes, PlaceType.VEHICLE_RENT);
     includeCarParking = shouldInclude(filterByPlaceTypes, PlaceType.CAR_PARK);
     includeBikeParking = shouldInclude(filterByPlaceTypes, PlaceType.BIKE_PARK);
+    includeStations =
+      shouldInclude(filterByPlaceTypes, PlaceType.STATION) &&
+      filterByPlaceTypes != null &&
+      !filterByPlaceTypes.isEmpty();
```

Comment:
It's not the most clear what is happening here. I think it would improve the readability if the
```
filterByPlaceTypes != null &&
      !filterByPlaceTypes.isEmpty();
```
would be either hidden behind some function call or parameter to a function call. Maybe we should just extend the shouldInclude function with a parameter called something like `defaultOn` but I think we try to avoid changing the behaviour of functions with boolean flags. Maybe we should have a Set of default filters something like Set<PlaceType> DEFAULTS = Set.of(PlaceType.VEHICLE_RENT, ...) and check from there within the shouldInclude function if the filter should be included.

Reply:
We can also shift this problem into the API layer.

Reply:
I think it could make sense. The same code is also used by the transmodel API I think and if we did some changes to the defaults for one API, we don't want to necessarily make the default changes to all APIs.

Reply:
So it would be fine to remove the null and emptiness checks and require `filterByPlaceTypes` to be nonnull in the constructor?

Reply:
I think so

Reply:
Simply requiring that `filterByPlaceFilter` is not null leads to some tests failing because they provide null for this parameter. I could change that parameter in those tests to empty list to make them pass but then the `shouldInclude` function should be changed as well to check for list emptiness. Would such a change be ok? It was said above that such changes should be avoided.

Reply:
I'm vote for not allowing nulls but I will let @optionsome have the last word.

Reply:
I'm OK with this. Null and empty list are then not allowed and we provide a constant for the callers if then request doesn't specify the filters, ok?

Reply:
Still a bit unsure about the constant part - what would the constant be? Is there an example of this behavior somewhere else? Or would it be ok to throw an exception for an illegal argument or something similar?